src/extractor/CaretValueRuleExtractor.ts

```diff
@@ -112,38 +112,65 @@ export class CaretValueRuleExtractor {
       if (flatParent[key] == null || !isEqual(flatSD[key], flatParent[key])) {
         const caretValueRule = new ExportableCaretValueRule('');
         caretValueRule.caretPath = key;
-        caretValueRule.value = getFSHValue(
-          key,
-          flatSD[key],
-          fhir.fishForFHIR('StructureDefinition', utils.Type.Resource),
-          fhir
-        );
+        caretValueRule.value = getFSHValue(key, flatSD[key], 'StructureDefinition', fhir);
         caretValueRules.push(caretValueRule);
       }
     });
 
     return caretValueRules;
   }
+
+  static processResource(
+    input: any,
+    fhir: fhirdefs.FHIRDefinitions,
+    resourceType: 'ValueSet' | 'CodeSystem'
+  ): ExportableCaretValueRule[] {
+    const caretValueRules: ExportableCaretValueRule[] = [];
+    const flatVS = getPathValuePairs(input);
+    Object.keys(flatVS)
+      .filter(
+        key => !RESOURCE_IGNORED_PROPERTIES[resourceType].some(property => key.startsWith(property))
+      )
+      .forEach(key => {
+        const caretValueRule = new ExportableCaretValueRule('');
+        caretValueRule.caretPath = key;
+        caretValueRule.value = getFSHValue(key, flatVS[key], resourceType, fhir);
+        caretValueRules.push(caretValueRule);
+      });
+    return caretValueRules;
+  }
 }
 
-// The properties that are handled via keywords and other extractors
-const IGNORED_PROPERTIES = [
-  'resourceType',
-  'id',
-  'url', // NOTE: For now, we assume URL matches canonical and is generated
-  'name',
-  'title',
-  'description',
-  'fhirVersion',
-  'mapping',
-  'baseDefinition',
-  'derivation',
-  'snapshot',
-  'differential'
-];
+const RESOURCE_IGNORED_PROPERTIES = {
+  ValueSet: [
+    'resourceType',
+    'id',
+    'url',
+    'name',
+    'title',
+    'description',
+    'compose.include',
+    'compose.exclude'
+  ],
+  CodeSystem: ['resourceType', 'id', 'url', 'name', 'title', 'description', 'concept'],
```

Comment:
Are we still planning for CodeSystems that have extra elements on a `concept` that aren't handled by other rules to be exported as FSH Instances? If so, this is fine, but if not, there might need to be some additional logic for getting those extra elements captured as CaretValueRules. And, similar question, but for ValueSet.

Reply:
@mint-thompson -- yes, that is the plan.  If we detect any unsupported properties on `concept` then we should generate an `Instance` representation instead.  I have a few stories assigned to me that start to get this underway, but I don't think I'll get to them this sprint.